src/hooks/useDataTrackPublisher/usePublishDataTrack.tsx

```diff
@@ -0,0 +1,44 @@
+import { TRACK_TYPE } from '../../utils/displayStrings';
+import { LocalDataTrack, LocalDataTrackOptions, LocalParticipant } from 'twilio-video';
+import useVideoContext from 'hooks/useVideoContext/useVideoContext';
+import { useState } from 'react';
+import isModerator from 'utils/rbac/roleChecker';
+import { ParticipantInformation } from 'state';
+
+export default function usePublishDataTrack(participantInfo: ParticipantInformation | null) {
+  const {
+    room: { localParticipant },
+  } = useVideoContext();
+
+  const [havePublishedLocalModeratorDataTrack, setHavePublishedLocalModeratorDataTrack] = useState(false);
+
+  if (havePublishedLocalModeratorDataTrack || participantInfo === null || localParticipant === undefined) {
+    return;
+  }
+
+  if (isModerator(participantInfo.partyType)) {
+    publishDataTrack(localParticipant);
+  }
+  setHavePublishedLocalModeratorDataTrack(true);
+}
+
+async function publishDataTrack(localParticipant: LocalParticipant) {
+  console.log('adding data track for this participant');
+
+  let localDataTrackOptions = {} as LocalDataTrackOptions;
+  //localDataTrackOptions.maxPacketLifeTime = 1000;
+  localDataTrackOptions.maxRetransmits = 3;
+  localDataTrackOptions.ordered = true;
+
+  var moderatorTrack = new LocalDataTrack(localDataTrackOptions);
```

Comment:
need to rename to localDataTrack. The word moderator should not really appear because the track is general to all dataTrack messages